text/1005-edgedb-project.rst

```diff
@@ -0,0 +1,418 @@
+and the ``instance-name`` file contains the name of the associated instance.
+
+Language bindings and clients may cache the resolved instance name in memory
+to avoid performing a lookup on every ``connect()`` call.  The cache should
+be sensitive to the current working directory, although if the directory
+changed to a subdirectory the cache should still be valid as nested projects
+are not allowed.
+
+Running ``project init`` from ``curl | sh``
+-------------------------------------------
+
+The CLI bootstrap script can detect if it's being ran from within a project
+directory and ask if the user wants to run ``edgedb project init``.
+
+Effect on ``edgedb server`` commands
+------------------------------------
+
+The ``edgedb server destroy`` command should refuse to destroy an instance that
+is associated with a project by default, and should recommend to use
+``edgedb project deinit`` (or ``edgedb server destroy --force``).
+
+The ``edgedb server upgrade`` command should refuse to continue if the target
+server version does not match the ``edgedb.server-version`` range in
+``edgedb.toml``.
+
+
+edgedb project init
+===================
+
+Initialize a new project or re-initialize an existing project.
+
+Synopsis
+--------
+
+``edgedb project init [options]``
+
+Options
+-------
+
+``--project-dir=<dir>``
+  Specifies a project root directory explicitly.  If not specified, the project
+  directory is detected as described in the "Project detection" section above,
+  and if no project directory is detected, a current working directory is used.
+
+``--server-version=<semver-range>``
+  Specifies the desired EdgeDB server version as a SemVer range.  Only
+  applicable for new projects.  Defaults to ``'*'``, which means latest stable
+  version for new projects.  Accepts ``'nightly'`` as a special value denoting
+  the latest nightly version.
+
+``--server-instance=<instance>``
+  Specifies the EdgeDB server instance to be associated with the project.
+  If the specified instance does not exist, it will be created.  If the
+  specified instance already exists, it must not be associated with another
+  project.  ``edgedb server deinit`` may be used to disassociate an instance
+  prior to linking it with another project.
+
+``--server-instance-type=<instance-type>``
+  Specifies the desired instance type.  Allowed values for ``<instance-type>``
+  are: ``native`` and ``docker``, which correspond to the installation modes
+  in ``edgedb server``.
+
+``--non-interactive``
+  Run in non-interactive mode.
+
+Implementation
+--------------
+
+The ``edgedb project init`` command initializes a brand new project or
+re-initializes an existing project.
+
+In a new project:
+
+- an ``edgedb.toml`` file is created in the project directory,
+  and ``--server-version``, if specified, is recorded in the
+  ``edgedb.server-version`` attribute.
+
+- a ``dbschema`` directory and a ``dbschema/default.esdl`` file are created,
```

Comment:
How "new project" is detected? Is it edgedb.toml? It's unspecified what we do if there is no `edgedb.toml`, but there are something in `dbschema`

Reply:
> there is no `edgedb.toml`, but there are something in `dbschema`

Presence of `dbschema` no longer indicates a "project", only `edgedb.toml`.  If `dbschema` already exists, we simply don't attempt to create it.  I'll clarify.

Reply:
Would also be cool to be able to configure a different dir for `dbschema` in `edgedb.toml`. For projects that use multiple databases simultaneously, `dbschema` might already be taken or just be confusing.

Reply:
Added